scripts/run-golangci.sh

```
@@ -40,16 +40,28 @@
+# worktree; the detector below stays armed for a caller who points
+# GOLANGCI_LINT_CACHE somewhere shared anyway. Clearing a stale cache remains
+# the human's call, for the reason it always was: destructive to a resource
+# other sessions may be using.
 set -euo pipefail
 
 root="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
 
+if [[ -z "${GOLANGCI_LINT_CACHE:-}" ]]; then
+  # cksum, not shasum: POSIX, present everywhere this runs. The basename rides
+  # along so a human listing the cache directory can tell the partitions apart.
+  export GOLANGCI_LINT_CACHE="$HOME/.cache/golangci-lint/$(basename "$root")-$(printf '%s' "$root" | cksum | cut -d' ' -f1)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)